torch/csrc/jit/init.cpp

```diff
@@ -117,14 +118,44 @@ void initJITBindings(PyObject* module) {
           [](std::shared_ptr<Graph>& g) { return PropagateQuantInfo(g); })
       .def(
           "_jit_pass_insert_observers",
-          [](std::shared_ptr<Graph>& g, py::function pyObserverFunction) {
+          [](std::shared_ptr<script::Module>& moduleObj,
+              py::dict& pyModuleObjDict) {
             // Create a new node that would be used in the insert observer pass:
             // all observer nodes will be cloned from this one.
-            Node* new_node = g->createPythonOp(
-                THPObjectPtr(pyObserverFunction.release().ptr()), "dd", {});
-            InsertObserverNodes(g, new_node);
-            // We don't need this node anymore, don't forget to remove it.
-            new_node->destroy();
+            if (!pyModuleObjDict.size()) {
+              return;
+            }
+            auto moduleObjDict = py::cast<std::unordered_map<std::string,
+              std::unordered_map<std::string,
+              py::object>>>(pyModuleObjDict);
+
+            // Lookup for the top level module quantObject
+            if (!moduleObjDict.count(moduleObj->name())) {
+              return;
+            }
+            auto& quantConfigObjDict = moduleObjDict[moduleObj->name()];
+            if (!quantConfigObjDict.size()) {
+              return;
+            }
+            script::Method* method = moduleObj->find_method("forward");
+            AT_ASSERT(method != nullptr);
+
+            auto g = method->graph();
+            // Convert to observer dict with new (key value)
+            std::unordered_map<std::string, Node*> observerNodeDict;
+            for (auto& it : quantConfigObjDict) {
```

Comment:
Please do not loop over unordered dictionaries and use the results to create graphs. It will put nodes in a non-determinsitic order, meaning output code is different on every run. It's not easy to debug!